### PR TITLE
initial metrics-rusage package

### PR DESCRIPTION
**Pull request by hannesm**
this gathers resource usage of the current process via getrusage, and also uptime etc. from /proc/self/stat and /proc/self/statm.

initial code from https://github.com/roburio/albatross, /cc @reynir

### CODE REVIEW

File: metrics-rusage.opam [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+maintainer:   "thomas@gazagnaire.org"
+authors:      ["Thomas Gazagnaire"]
+license:      "ISC"
+homepage:     "https://github.com/mirage/metrics"
+bug-reports:  "https://github.com/mirage/metrics/issues"
+dev-repo:     "git+https://github.com/mirage/metrics.git"
+doc:          "https://mirage.github.io/metrics/"
+
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (avsm) [inline metrics-rusage.opam:11]:
```suggestion
  ["dune" "subst"] {dev}
```

See https://github.com/ocaml/opam-repository/pull/18732

File: src/rusage/metrics_rusage.ml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+let tv (sec, usec) = Int64.to_float sec +. float_of_int usec /. 1_000_000.
+
+open Metrics
+
+let rusage_src ~tags =
+  let doc = "System rusage counters" in
+  let graph = Graph.v ~title:doc ~ylabel:"value" () in
+  let data () =
+    match wrap getrusage () with
```
Comment (avsm) [inline src/rusage/metrics_rusage.ml:117]: Can getrusage really return an EINTR (and thus require `wrap`)?

https://man7.org/linux/man-pages/man2/getrusage.2.html doesnt have EINTR in its return list

File: metrics-rusage.opam (hunk 1 above)
Comment (hannesm) [reply metrics-rusage.opam]: fixed, also in all the other opam files.

File: src/rusage/metrics_rusage.ml (hunk 2 above)
Comment (hannesm) [reply src/rusage/metrics_rusage.ml:103]: that is true, fixed in db06c51

**Pull request merged**